src/main/java/digital/slovensko/autogram/server/dto/SignRequestBody.java

```diff
@@ -47,10 +56,34 @@ public SigningParameters getParameters() throws RequestValidationException {
 
         parameters.validate(getDocument().getMimeType());
 
-        return parameters.getSigningParameters(isBase64());
+        SigningParameters signingParameters = parameters.getSigningParameters(isBase64());
+
+        if (!validateXml(getDecodedContent(), signingParameters.getSchema()))
+            throw new RequestValidationException("XML validation against XSD failed", "");
+
+        return signingParameters;
+    }
+
+    private boolean validateXml(String xmlContent, String xsdSchema) {
+        if (xsdSchema == null) {
+            return true;
+        }
+        try {
+            SchemaFactory factory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
+            Schema schema = factory.newSchema(new StreamSource(new StringReader(xsdSchema)));
+            Validator validator = schema.newValidator();
+            validator.validate(new StreamSource(new StringReader(xmlContent)));
```

Comment:
Skus preklikat vsetky ukazky v swaggeri, ze ci to nahodou nefailuje niekde.

Reply:
V 2 prikladoch neprejde validacia:
- XAdES (ASiC_E) Slovak XML Data Container with Base64-encoded XML parameters and HTML transformation
- XAdES (ASiC_E) Slovak XML Data Container with plaintext XML parameters and HTML transformation

exception: 
- cvc-elt.1.a: Cannot find the declaration of element 'xdc:XMLDataContainer'.

Xml v prikladoch je vzdy validne podla schemy, alebo sa moze stat ze je tam realne chyba? 
@jsuchal

Reply:
@Bombino1024 XML datacontainer je taká obálka okolo pôvodného xml. Treba najprv vybrať pôvodné xml z toho a až to validovať. Pôvodné xml je v tagu `xdc:XMLData`. Pošlem ti na Slacku nejaké súbory.

Tým pádom nebudeme validovať samotnú XDC obálku. V nej je ale ešte uložený hash XSD a XSLT, takže toto by ešte asi bolo treba tiež validovať, či sedí. Aby sa nestalo, že človek dostane zmenenú transformáciu, ktorá ho oklame v náhľade, a podpíše niečo iné, ale v XDC bude zahashovaná pôvodná správna transformácia a teda s tým na súde nič nespraví.

Reply:
@Bombino1024 to co pise @celuchmarek je tu zapracovane?